python/foxglove-sdk/python/docs/api/index.rst

```diff
@@ -3,12 +3,77 @@ API Reference
 
 Version: |release|
 
+.. note::
+   The notebook integration classes and functions are only available when the `notebook` extra package is installed.
+   Install it with `pip install foxglove-sdk[notebook]`.
+
 foxglove
 --------
 
 .. automodule:: foxglove
    :members:
-   :exclude-members: MCAPWriter
+   :exclude-members: MCAPWriter, init_notebook_buffer
+
+Notebook Integration
+^^^^^^^^^^^^^^^^^^^^
+
+Functions and classes for integrating with Jupyter notebooks and creating interactive visualizations.
+
+.. py:function:: init_notebook_buffer(context: Context | None = None) -> NotebookBuffer
+
+   Create a NotebookBuffer object to manage data buffering and visualization in Jupyter
+   notebooks.
+
+   :param context: The Context used to log the messages. If no Context is provided, the global
+       context will be used. Logged messages will be buffered.
+   :return: A NotebookBuffer object that can be used to manage the data buffering
+       and visualization.
+   :raises Exception: If the notebook extra package is not installed. Install it
+       with `pip install foxglove-sdk[notebook]`.
+
+   .. note::
+      This function is only available when the `notebook` extra package
+      is installed. Install it with `pip install foxglove-sdk[notebook]`.
+      The widget will automatically load the data from the provided datasource
+      and display it in the embedded Foxglove viewer.
+
+Notebook Classes
+^^^^^^^^^^^^^^^^
+
+.. py:class:: NotebookBuffer
+
+   A NotebookBuffer object to manage data buffering and visualization in Jupyter notebooks.
+
+   Obtain an instance by calling :py:func:`foxglove.init_notebook_buffer`.
+
+   The NotebookBuffer object will buffer all data logged to the provided context. When you
+   are ready to visualize the data, you can call the :meth:`NotebookBuffer.show` method to display an embedded
+   Foxglove visualization widget. The widget provides a fully-featured Foxglove interface
+   directly within your Jupyter notebook, allowing you to explore multi-modal robotics data
+   including 3D scenes, plots, images, and more.
+
+   .. py:method:: show(width: int | Literal["full"] | None = None, height: int | None = None, src: str | None = None, layout: SelectLayoutParams | None = None) -> FoxgloveWidget
+
+      Show the Foxglove viewer. Call this method as the last step of a notebook cell
+      to display the viewer.
+
+   .. py:method:: clear() -> None
+
+      Clear the buffered data.
+
+   .. py:method:: get_data() -> list[bytes]
+
+      Retrieve all collected data.
+
```

Comment:
why is this method exposed/documented? Are users expected to use it for something?